tests/pytests/test_dfjcosk.py

```diff
@@ -86,3 +85,60 @@ def test_dfjcosk(inp, mols):
     psi4.set_options({"scf_type" : "pk"})
     energy_pk = psi4.energy(inp["method"], molecule=molecule, bsse_type=inp["bsse_type"])
     assert compare_values(energy_pk, energy_dfjcosk, atol=1e-4)
+
+@pytest.mark.parametrize(
+    "inp",
+    [
+        pytest.param({"method" : "hf",
+                      "options": {"reference" : "rhf"},
+                      "molecule" : "h2o",
+                      "bsse_type" : None,
+                      "ref" : -76.026780223322},
+                      id="h2o (rhf)"),
+        pytest.param({"method" : "b3lyp",
+                      "options": {"reference" : "rhf"},
+                      "molecule" : "h2o",
+                      "bsse_type" : None,
+                      "ref" : -76.420402720419},
+                      id="h2o (rks)"),
+        pytest.param({"method" : "hf",
+                      "options": {"reference" : "uhf"},
+                      "molecule" : "nh2",
+                      "bsse_type" : None,
+                      "ref" : -55.566890252551},
+                      id="nh2 (uhf)"),
+        pytest.param({"method" : "hf",
+                      "options": {"reference" : "rohf"},
+                      "molecule" : "nh2",
+                      "bsse_type" : None,
+                      "ref" : -55.562689948780},
+                      id="nh2 (rohf)"),
+        pytest.param({"method" : "hf",
+                      "options": {"reference" : "rhf"},
+                      "molecule" : "h2o_nap1",
+                      "bsse_type" : "CP",
+                      "ref" :  -0.040121884077},
+                      id="h2o/na+ (rhf ie)"),
+    ],
+)
+def test_dfjcosk_incfock(inp, mols):
+    """Test the efficiency of IncFock in DFJCOSK JK object via SCF calculations"""
+
+    molecule = mols[inp["molecule"]]
+    psi4.set_options({"scf_type" : "cosx", "basis": "cc-pvdz", "incfock": False})
+    psi4.set_options(inp["options"])
+
+    # compute DFJCOSK energy+wfn without IncFock 
+    energy_dfjcosk_noinc, wfn_dfjcosk_noinc = psi4.energy(inp["method"], molecule=molecule, bsse_type=inp["bsse_type"], return_wfn=True)
+    #assert compare_values(inp["ref"], energy_dfjcosk, atol=1e-6)
+
+    # compute DFJCOSK energy+wfn with Incfock 
+    psi4.set_options({"incfock" : True})
+    energy_dfjcosk_inc, wfn_dfjcosk_inc = psi4.energy(inp["method"], molecule=molecule, bsse_type=inp["bsse_type"], return_wfn=True)
+
+    # how do energies compare?
+    assert compare_values(energy_dfjcosk_noinc, energy_dfjcosk_inc, atol=1e-6)
+    
+    # how do SCF iteration counts compare?
+    if hasattr(wfn_dfjcosk_noinc, "iteration_") and hasattr(wfn_dfjcosk_inc, "iteration_"):
+        assert compare_values(wfn_dfjcosk_noinc.iteration_, wfn_dfjcosk_inc.iteration_, 3)
```

Comment:
`niter_noinc = int(wfn_dfjcosk_noinc.variable("SCF ITERATIONS"))` can give you the count, too

I'm not sure what you're doing in the assert. why would the the niter be the same to 1.e-3? You could do `assert compare(True, (niter_noinc - niter_inc) >= 3, "reduced by at least 3 iterations by incfock")` if that's what you had in mind.

Reply:
What I was trying to do with the test was to make sure that the SCF iterations for the IncFock run was not significantly higher than that of the non-IncFock run. IncFock won't generally reduce the number of SCF iterations significantly, but it CAN greatly increase them. This test is designed to ensure that the IncFock run doesn't require significantly more iterations (defined as more than 3 in this test); and this assert is the point at which the comparison is made. 

Thank's for the clarification on the compare functions!

Reply:
Ah, I see now, thanks. do you want an `abs(niter_noinc - niter_inc)` then?

Reply:
You're welcome! Abs could be something we do, because it would catch cases where IncFock gives significantly lower SCF iterations than no-IncFock. It may be worth catching this in case it indicates some other error.

Reply:
All right, I think this file has been largely refined based on the discussion here!